src/peakheight.jl

```diff
@@ -1,67 +1,111 @@
-        maxheight=nothing
-    ) -> (peaks, heights)
+    peakheights(indices, heights; [min, max]) -> (indices, heights)
+    peakheights(pks::NamedTuple; [min, max]) -> NamedTuple
 
-Return a copy of `peaks` and `heights` where peak heights are removed if less than
-`minheight` and/or greater than `maxheight`.
 
-See also: [`peakprom`](@ref), [`peakwidths`](@ref), [`findmaxima`](@ref)
+Return a copy of `indices` and `heights` where peaks are removed if their height is less than
+`min` and/or greater than `max`.
+
+If a NamedTuple `pks` is given, a new NamedTuple is returned with filtered copies of the
+fields in `pks`. `pks` must have `:indices` and `:heights` fields, at a minimum.
+The following fields will also be copied and filtered if they exist: `:proms`, `:widths`,
+and `:edges`.
+
+See also: [`peakprom`](@ref), [`peakwidths`](@ref), [`findmaxima`](@ref),
+[`filterpeaks`](@ref)
 
 # Examples
 ```jldoctest
 julia> x = [0,5,2,3,3,1,4,0];
 
-julia> xpks, vals = findmaxima(x)
-([2, 4, 7], [5, 3, 4])
+julia> indices, heights = findmaxima(x)
+(indices = [2, 4, 7], heights = [5, 3, 4], data = [0, 5, 2, 3, 3, 1, 4, 0])
 
-julia> peakheights(xpks, vals; maxheight=4)
+julia> peakheights(indices, heights; max=4)
 ([4, 7], [3, 4])
 
-julia> peakheights(xpks, vals; minheight=4.5)
-([2], [5])
+julia> peakheights((;indices, heights, data=x); min=4)
+(indices = [2], heights = [5], data = [0, 5, 2, 3, 3, 1, 4, 0])
 ```
 """
 function peakheights(
-    peaks::AbstractVector{Int}, heights::AbstractVector;
-    minheight=nothing, maxheight=nothing
+    indices::AbstractVector{Int}, heights::AbstractVector;
+    minheight=nothing, maxheight=nothing,
+    min=minheight, max=maxheight
 )
-    peakheights!(copy(peaks), copy(heights); minheight=minheight, maxheight=maxheight)
+    if !isnothing(minheight)
+        Base.depwarn("Keyword `minheight` has been renamed to `min`", :peakheights!)
+    end
+    if !isnothing(maxheight)
+        Base.depwarn("Keyword `maxheight` has been renamed to `max`", :peakheights!)
+    end
+    peakheights!(copy(indices), copy(heights); min=min, max=max)
 end
 
+peakheights(pks::NamedTuple; kwargs...) = peakheights!(deepcopy(pks); kwargs...)
+
 """
-    peakheights!(peaks, heights;
-        minheight=nothing,
-        maxheight=nothing
-    ) -> (peaks, heights)
+    peakheights(; min=nothing, max=nothing) -> Function
 
-Modify and return `peaks` and `heights` by removing peaks that are less than `minheight` or greater
-than `maxheight`.
+Create a function that filters the fields of a copy of the singular NamedTuple argument
+`pks`.
 
-See also: [`peakprom`](@ref), [`peakwidths`](@ref), [`findmaxima`](@ref)
+This allows for piping several `Peaks.jl` functions together.
+
+## Examples
+```julia-repl
+julia> pks |> peakheights(;min=4)
+(indices = [2], heights = [5], data = [1, 5, 1, 3, 2])
+```
+"""
+peakheights(; kwargs...) = function curried_peakheights(pks)
+    return peakheights(deepcopy(pks); kwargs...)
+end
```

Comment:
Reproduce the name function style wherever else is appropriate; the named function will improve any stacktraces it appears in (better than "#15 (generic function" imo)

Reply:
Great improvement (Y)